docs/website/docs/dlt-ecosystem/verified-sources/rest_api.md

```diff
@@ -0,0 +1,527 @@
+---
+title: REST API generic source
+description: dlt verified source for REST APIs
+keywords: [rest api, restful api]
+---
+import Header from './_source-info-header.md';
+
+<Header/>
+
+This is a generic dlt source you can use to extract data from any REST API. It uses declarative configuration to define the API endpoints, their relationships, parameters, pagination, and authentication.
+
+## Setup guide
+
+### Initialize the verified source
+
+Enter the following command in your terminal:
+
+```sh
+dlt init rest_api duckdb
+```
+
+[dlt init](../../reference/command-line-interface) will initialize the pipeline examples for REST API as the [source](../../general-usage/source) and [duckdb](../destinations/duckdb.md) as the [destination](../destinations).
+
+Running `dlt init` creates the following in the current folder:
+- `rest_api_pipeline.py` file with a sample pipelines definition:
+    - GitHub API example
+    - Pokemon API example
+- `.dlt` folder with:
+     - `secrets.toml` file to store your access tokens and other sensitive information
+     - `config.toml` file to store the configuration settings
+- `requirements.txt` file with the required dependencies
+
+Change the REST API source to your needs by modifying the `rest_api_pipeline.py` file. See the detailed [source configuration](#source-configuration) section below.
+
+:::note
+For the rest of the guide, we will use the [GitHub API](https://docs.github.com/en/rest?apiVersion=2022-11-28) and [Pokemon API](https://pokeapi.co/) as example sources.
+:::
+
+This source is based on the [RESTClient class](../../general-usage/http/rest-client.md).
+
+### Add credentials
+
+In the `.dlt` folder, you'll find a file called `secrets.toml`, where you can securely store your access tokens and other sensitive information. It's important to handle this file with care and keep it safe.
+
+The GitHub API [requires an access token](https://docs.github.com/en/rest/authentication/authenticating-to-the-rest-api?apiVersion=2022-11-28) to access some of its endpoints and to increase the rate limit for the API calls. To get a GitHub token, follow the GitHub documentation on [managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+
+After you get the token, add it to the `secrets.toml` file:
+
+```toml
+[sources.rest_api.github]
+github_token = "your_github_token"
+```
+
+## Run the pipeline
+
+1. Install the required dependencies by running the following command:
+
+   ```sh
+   pip install -r requirements.txt
+   ```
+
+2. Run the pipeline:
+
+   ```sh
+    python rest_api_pipeline.py
+    ```
+
+3. Verify that everything loaded correctly by using the following command:
+
+    ```sh
+    dlt pipeline rest_api show
+    ```
+
+## Source configuration
+
+### Quick example
+
+Let's take a look at the GitHub example in `rest_api_pipeline.py` file:
+
+```py
+def load_github() -> None:
```

Comment:
this is technically correct but IMO not the patter we show our users in other places. the pattern is to put resources in dlt.source and use it to pass additional configuration and secrets

so the code below
```
@dlt.source
def githiub_source(github_token=dlt.secrets.value):
      yield from rest_api_resource(config)
```
and then pipeline.run

btw. then your example above:
```toml
[sources.rest_api.github]
github_token = "your_github_token"
```
will also work because now the config structure conforms to how we organize source secrets

Reply:
Updated in [cafa96c](https://github.com/dlt-hub/dlt/pull/1308/commits/cafa96c542d1eeb14f85bbfd4a2affddbf0a9a73)